pom.xml

```diff
@@ -48,6 +48,9 @@
                   <outputFormat>XML</outputFormat>
                   <outputFormat>HTML</outputFormat>
               </outputFormats>
+              <excludedMethods>
+              	<excludedMethod>hashCode</excludedMethod>
+              </excludedMethods>
```

Comment:
The only functional requirement of hashCode it to return the same value for equal objects, therefore mutations that replaces return value to constant or (x == 0 ? 1 : 0) like expression generates a valid hashCode function.

Reply:
then test that functional requirement
- the hashcode of an entity obtained in different times is the same
- the hashcode of two different entities is different

Reply:
The general contract of hashCode() states:

* Whenever it is invoked on the same object more than once during an execution of a Java application, hashCode() must consistently return the same value, provided no information used in equals comparisons on the object is modified. This value needs not remain consistent from one execution of an application to another execution of the same application
* If two objects are equal according to the equals(Object) method, then calling the hashCode() method on each of the two objects must produce the same value
* It is not required that if two objects are unequal according to the equals(java.lang.Object) method, then calling the hashCode method on each of the two objects must produce distinct integer results. However, developers should be aware that producing distinct integer results for unequal objects improves the performance of hash tables

The 1st of your requirements is valid only within 1 execution of the application. If we change the hashCode implementation, it can be changed, therefore we can't test obj.hashCode() == CONST.

The 2nd one is not true. There is no requirement that tells that two objects with different class must have different hashCode.

Reply:
The 2nd one is not required, but recommended anyway.
I have checked, and the zombie dies if you make a test which makes sure that two different entities have different hashcode.

Reply:
I don't like this solution, but 6 Euro is 6 Euro :laughing:

Reply:
And I don't like exclusions from code analysis. PIT is not even false positive on it.